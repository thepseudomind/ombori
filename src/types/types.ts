export interface IMain{
    status: boolean,
    toggleSidebar?: any
}

export interface IProduct{
    id: number,
    email: string,
    first_name: string,
    last_name: string,
    avatar: string
}

export type TLoader = {
    mini: boolean
}

export type TMessage = {
    message: string
}