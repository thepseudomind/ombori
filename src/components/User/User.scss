@import '../../mixins/mixins';

.user{
    height: 30rem;
    width: 25rem;
    margin: 2rem;
    border-radius: .3rem;
    overflow: hidden;
    box-shadow: 0 .1rem 5rem rgba(black, .1);
    font-family: 'Barlow Semi Condensed', sans-serif;
    cursor: pointer;

    @include respond(tab-land){
        width: 28rem;
    }

    @include respond(phone){
        width: 27.5rem;
    }

    &.active{
        width: 27.5rem;
    }

    &__header{
        height: 50%;
        color: white;
        background-color: black;
        display: flex;
        justify-content: center;
        align-items: center;

        &--avatar{
            height: 10rem;
            width: 10rem;
            border-radius: 50%;
        }
    }

    &__content{
        height: 50%;
        width: 80%;
        margin: 0 auto;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        font-size: 1.2rem;
        font-weight: 400;
        text-transform: uppercase;

        & > *{
            &:not(:last-child){
                margin-bottom: .5rem;
            }
        }
    }

    &__name{
        font-family: 'Barlow Semi Condensed', sans-serif;
        font-size: 3rem;
        font-weight: 700;
        text-align: center;
    }

    &__email{
        text-align: center;
    }
}