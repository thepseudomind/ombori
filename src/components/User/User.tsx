import React from 'react';
import './User.scss';
import { IProduct } from '../../types/types';

const User =({id, email, first_name, last_name, avatar} : IProduct)=>{
    return (
        <div className="user">
            <div className="user__header">
                <img src={avatar} alt={`${first_name}-${id}`} className="user__header--avatar"/>
            </div>
            <div className="user__content">
                <p className="user__name">{`${first_name} ${last_name}`}</p>
                <p className="user__email">{email}</p>
            </div>
        </div>
    );
}

export default User;