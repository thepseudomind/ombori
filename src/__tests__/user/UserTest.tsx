import React from 'react';
import { shallow } from  'enzyme';
import User from '../../components/User/User';
import { IProduct } from '../../types/types';

describe('user test', ()=>{
    const mockProps : IProduct = {
        id: 1,
        email: 'email@email.com',
        first_name: 'Name',
        last_name: 'Other name',
        avatar: 'link'
    }

    const wrapper = shallow(<User {...mockProps}/>)

    it('render user', ()=>{
        expect(wrapper.length).toEqual(1);
        expect(wrapper).toMatchSnapshot();
    });
})
