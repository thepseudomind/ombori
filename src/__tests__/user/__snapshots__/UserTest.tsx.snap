// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user test shallow render 1`] = `
<div
  className="user"
>
  <div
    className="user__header"
  >
    <img
      alt="Name-1"
      className="user__header--avatar"
      src="link"
    />
  </div>
  <div
    className="user__content"
  >
    <p
      className="user__name"
    >
      Name Other name
    </p>
    <p
      className="user__email"
    >
      email@email.com
    </p>
  </div>
</div>
`;
